sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BeamBigQueryTable.java

```diff
@@ -43,7 +52,30 @@ public BeamBigQueryTable(Schema beamSchema, String tableSpec) {
 
   @Override
   public PCollection<Row> buildIOReader(PBegin begin) {
-    throw new UnsupportedOperationException();
+    return begin
+        .apply(
+            BigQueryIO.read(
+                    new SerializableFunction<SchemaAndRecord, Row>() {
```

Comment:
This should definitely live function in `sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java`

Reply:
If I should wrap it into `BigQueryUtils.java`, then `AvroUtils.java` should also be in IO component so I don't need add dependency of SQL to IO (as IO will need call AvroUtils if not do so).

Reply:
I believe `AvroUtils.java` is going to end up in something like `org.apache.beam.sdk.schemas.utils`, but until you hear from Reuven where it should go the Bigquery IO is a good location.